qcfractal/tests/test_collections.py

```diff
@@ -1157,7 +1157,7 @@ def test_reactiondataset_dftd3_dataset_plaintextview_write(reactiondataset_dftd3
 def test_collection_query(fractal_compute_server):
     client = ptl.FractalClient(fractal_compute_server)
 
-    ds = ptl.collections.Dataset("CAPITAL", client)
+    ds = ptl.collections.Dataset("CAPITAL", client, owner="default")
```

Comment:
If it's the `owner="default"`, then we shouldn't specify it in the call., it should be the default.

Reply:
It's only the default if the client is not logged in.

Reply:
Even If I'm logged in, I won't like to write this in every query. I should always get the same behavior and use the same code either when I'm logged in or not and get the same results.

Reply:
I'm not really a fan of having this set as the logged in user as it could lead to confusion. What if we change this to `group` to remove the single-user ownership notion and give us grounds for integration into the groups framework that we sort of have planned?

Reply:
Per discussion with @dgasmith, we will:
- change owner to group
- the default value for group for new datasets will be "default"
- the default value for group for client.list_collections will be "default"
- there will be no username-related funny business